libgpos/include/gpos/memory/CMemoryPoolCustomAlloc.h

```diff
@@ -0,0 +1,95 @@
+//---------------------------------------------------------------------------
+//	Greenplum Database
+//	Copyright (c) 2004-2015 Pivotal Software, Inc.
+//
+//	@filename:
+//		CMemoryPoolCustomAlloc.h
+//
+//	@doc:
+//		Implementation of memory pool that allocates memory by calling a custom allocator
+//
+//	@owner:
+//
+//	@test:
+//
+//---------------------------------------------------------------------------
+#ifndef GPOS_CMemoryPoolCustomAlloc_H
+#define GPOS_CMemoryPoolCustomAlloc_H
+
+#include "gpos/base.h"
+#include "gpos/memory/CAutoMemoryPool.h"
+
+namespace gpos
+{
+	// memory pool wrapping clib allocation API
+	class CMemoryPoolCustomAlloc : public CMemoryPool
+	{
+		private:
+
+			// private copy ctor
+			CMemoryPoolCustomAlloc(CMemoryPoolCustomAlloc &);
+			void* (*PvCustomAlloc) (SIZE_T);
+			void (*CustomFree) (void*);
+
+		public:
+
+			// ctor
+			CMemoryPoolCustomAlloc
+				(
+					void* (*fnAlloc)(SIZE_T),
```

Comment:
pfAlloc or pfuncAlloc
always start with p, meaning it is a pointer to something